packages/SwingSet/src/lib/kmarshal.js

```diff
@@ -0,0 +1,55 @@
+import { Far, makeMarshal } from '@endo/marshal';
+
+// Simple wrapper for serializing and unserializing marshalled values inside the
+// kernel, where we don't actually want to use clists nor actually allocate real
+// objects, but instead to stay entirely within the domain of krefs.  This is
+// used to enable syntactic manipulation of serialized values while remaining
+// agnostic about the internal details of the serialization encoding.
+
+const promiseMap = new WeakMap();
+
+export const kslot = (kref, iface) => {
```

Comment:
On second thought, seeing how this is used to build placeholders (e.g. when building the bootstrap message), I'm not sure `convertKslot` is ideal. I don't know what to recommend, so I guess I'd recommend leaving it as is.

Reply:
It is used as the `convertSlotToVal` function for the kernel marshaller, but that's one use out of more than 300.  As I said above, a better name would be good, but I haven't yet thought of one.  The main thing is that the name needs to be short since this is used so so much.